### PR TITLE
Add foreign keys when creating a new transition table

### PR DESCRIPTION
**Pull request by greysteil**
Because `Statesman is ... designed to provide ... data integrity.`. :octocat:

### CODE REVIEW

File: lib/generators/statesman/templates/create_migration.rb.erb [hunk 1]
```diff
@@ -9,6 +9,8 @@ class Create<%= migration_class_name %> < ActiveRecord::Migration
       t.timestamps null: false
     end
 
+    add_foreign_key :<%= table_name %>, :<%= parent_table_name %>
```
Comment (kenchan0130) [inline lib/generators/statesman/templates/create_migration.rb.erb]: I think that you should change `lib/generators/statesman/templates/update_migration.rb.erb` too.

**Comment by kenchan0130** [comment]
I think that introducing foreign key constraints depends on team and project policies.
Is it more appropriate to put it in the template as a position of this gem?

File: lib/generators/statesman/templates/create_migration.rb.erb (hunk 1 above)
Comment (greysteil) [inline lib/generators/statesman/templates/create_migration.rb.erb]: I deliberately kept it out of there, under the assumption that the end user would have already added a foreign key to the existing table they're updating if they wanted one.

**Comment by greysteil** [comment]
@kenchan0130 - good point, I've updated the migration to include a comment above the foreign key addition.

File: lib/generators/statesman/templates/create_migration.rb.erb (hunk 1 above)
Comment (kenchan0130) [inline lib/generators/statesman/templates/create_migration.rb.erb]: > under the assumption that the end user would have already added a foreign key to the existing table they're updating if they wanted one.

I see, but it depends on the situation.

**Comment by kenchan0130** [comment]
@greysteil 

Your comment approach is working out a compromise.
look to me good.

**Comment by Sinjo** [comment]
Agree with having them by default in the template. If people don't like foreign keys they can always delete them from the generated migration.

**Comment by Sinjo** [comment]
An afterthought: I hadn't thought about `add_foreign_key` with SQLite. It turns out it's just a no-op there.

I'm not sure if we care about SQLite support (we don't test it). I'll have a chat with another maintainer or two and either add SQLite to the build or add an explicit notice in `README.md` that we only support Postgres and MySQL.

**Pull request merged**